gk/main.c

```diff
@@ -96,6 +487,13 @@ run_gk(struct gk_config *gk_conf)
 	if (!gk_conf)
 		return -1;
 
+	gk_conf->num_lcores = gk_conf->lcore_end_id - gk_conf->lcore_start_id + 1;
+	gk_conf->instances = (struct gk_instance *)rte_calloc(NULL,
+		gk_conf->num_lcores, sizeof(struct gk_instance), 0);
```

Comment:
We clean up here. We should clean "everywhere". The only exception that I've opened was meant to avoid the Lua configuration to start to look like C code. I understand that this may looks confusing at this point, but I hope it'll become clearer as we get more code in Gatekeeper.